jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt

```diff
@@ -110,46 +118,75 @@ class DeployServerlessApplicationAction : AnAction(
     }
 
     private fun continueDeployment(project: Project, stackName: String, templateFile: VirtualFile, stackDialog: DeployServerlessApplicationDialog) {
-        val deployDialog = SamDeployDialog(
+        val workflow = StepExecutor(
             project,
-            stackName,
-            templateFile,
-            stackDialog.parameters,
-            stackDialog.bucket,
-            stackDialog.ecrRepo,
-            stackDialog.autoExecute,
-            stackDialog.useContainer,
-            stackDialog.capabilities
+            message("serverless.application.deploy_in_progress.title", stackName),
+            createDeployWorkflow(
+                project,
+                stackName,
+                templateFile,
+                stackDialog.bucket,
+                stackDialog.ecrRepo,
+                stackDialog.useContainer,
+                stackDialog.parameters,
+                stackDialog.capabilities
+            ),
+            stackName
         )
 
-        deployDialog.show()
-        if (!deployDialog.isOK) return
+        workflow.onSuccess = {
+            runBlocking {
+                val changeSetArn = it.getRequiredAttribute(DeployLambda.CHANGE_SET_ARN)
 
-        val cfnClient = project.awsClient<CloudFormationClient>()
+                if (!stackDialog.autoExecute) {
+                    val response = withContext(edtContext) { UploadFunctionContinueDialog(project, changeSetArn).showAndGet() }
+                    if (!response) {
+                        // TODO telemetry
```

Comment:
Did we never have telemetry here?

Reply:
Will clarify the message and fix telemetry here